nilearn/plotting/surf_plotting.py

```diff
@@ -29,13 +30,535 @@
+                "from avg_method does not have the correct shape: "
+                f"{surf_map_faces[0]} != {face_colors_size}")
+
+        # check that dtype is either int or float
+        if not (
+            "int" in str(surf_map_faces.dtype)
+            or "float" in str(surf_map_faces.dtype)
+        ):
+            raise ValueError(
+                'Array computed with the custom function '
+                'from avg_method should be an array of numbers '
+                '(int or float)'
+            )
+    else:
+        raise ValueError(error_message)
+    return surf_map_faces
+
+
+def _get_ticks_matplotlib(vmin, vmax, cbar_tick_format):
+    """Helper function for plot_surf with matplotlib engine.
+
+    This function computes the tick values for the colorbar.
+    """
+    # Default number of ticks is 5...
+    n_ticks = 5
+    # ...unless we are dealing with integers with a small range
+    # in this case, we reduce the number of ticks
+    if cbar_tick_format == "%i" and vmax - vmin < n_ticks:
+        ticks = np.arange(vmin, vmax + 1)
+        n_ticks = len(ticks)
+    else:
+        ticks = np.linspace(vmin, vmax, n_ticks)
+    return ticks
+
+
+def _compute_facecolors_matplotlib(bg_map, faces, n_vertices,
+                                   darkness, alpha):
+    """Helper function for plot_surf with matplotlib engine.
+
+    This function computes the facecolors.
+    """
+    # set alpha if in auto mode
+    if alpha == 'auto':
+        alpha = .5 if bg_map is None else 1
+    face_colors = np.ones((faces.shape[0], 4))
+    if bg_map is None:
+        bg_data = np.ones(n_vertices) * 0.5
+    else:
+        bg_data = load_surf_data(bg_map)
+        if bg_data.shape[0] != n_vertices:
+            raise ValueError('The bg_map does not have the same number '
+                             'of vertices as the mesh.')
+    bg_faces = np.mean(bg_data[faces], axis=1)
+    if bg_faces.min() != bg_faces.max():
+        bg_faces = bg_faces - bg_faces.min()
+        bg_faces = bg_faces / bg_faces.max()
+    # control background darkness
+    bg_faces *= darkness
+    face_colors = plt.cm.gray_r(bg_faces)
+    # modify alpha values of background
+    face_colors[:, 3] = alpha * face_colors[:, 3]
+    return face_colors
+
+
+def _threshold_and_rescale(data, threshold, vmin, vmax):
+    """Helper function for plot_surf.
+
+    This function thresholds and rescales the provided data.
+    """
+    data_copy, vmin, vmax = _rescale(data, vmin, vmax)
+    return data_copy, _threshold(data, threshold), vmin, vmax
+
+
+def _threshold(data, threshold):
+    # If no thresholding and nans, filter them out
+    return (np.logical_not(np.isnan(data)) if threshold is None
+            else np.abs(data) >= threshold)
+
+
+def _rescale(data, vmin, vmax):
+    data_copy = np.copy(data)
+    # if no vmin/vmax are passed figure them out from data
+    vmin, vmax = _get_bounds(data_copy, vmin, vmax)
```

Comment:
Just a minor change, but setting `vmin=None, vmax=None` by default could prove more handy in the future, and feels more consistent with other util functions.
```suggestion
def _rescale(data, vmin=None, vmax=None):
    data_copy = np.copy(data)
    # if no vmin/vmax are passed figure them out from data
    vmin, vmax = _get_bounds(data_copy, vmin, vmax)
```

Reply:
Thanks for the review @alexisthual ! :+1: